test.ts

```diff
@@ -1,60 +1,87 @@
+  duration: number;
+};
+
+export type ModelMetricParams = {
+  [K in Metric]: K extends keyof ModelBaseParams ? never : string;
+};
+
+export type ModelParams = ModelBaseParams | ModelMetricParams;
+
+export type AggregationParams = {
+  "aggregation-metrics": Metric[];
+  "aggregation-method": string;
+};
+
+export type AggregatedMetrics = Record<`aggregated-${Metric}`, number>
+
+const AVERAGES = ["av", "avg", "avrg", "average", "mean"];
+
+/**
+ * @todo Refactor this function according to your best practices.
+ * Input should stay the same. Feel free to introduce structure you like.
+ * Desired output should be
+ * {
+ *  aggregated-${aggregation-metrics[0]}: <calculated output here>
+ *  aggregated-${aggregation-metrics[1]}: <calculated output here>
+ *  ...
+ * }
+ */
+
+// return type
+export function execute(inputs: ModelParams[], params: AggregationParams): AggregatedMetrics {
   
-  type AggregationParams = {
-    'aggregation-metrics': string[];
-    'aggregation-method': string;
-  };
-  
-  /**
-   * @todo Refactor this function according to your best practices.
-   * Input should stay the same. Feel free to introduce structure you like.
-   * Desired output should be
-   * {
-   *  aggregated-${aggregation-metrics[0]}: <calculated output here>
-   *  aggregated-${aggregation-metrics[1]}: <calculated output here>
-   *  ...
-   * }
-   */
-  async function execute(inputs: ModelParams[], params: AggregationParams) {
-    if (inputs === undefined) {
-      throw new Error('Input data is missing');
-    }
-  
-    if (!Array.isArray(inputs)) {
-      throw new Error('Input data is not an array');
+  _validateInputParams(inputs, params);
+
+  const aggregates: AggregatedMetrics = {};
+
+  for (const metricName of params['aggregation-metrics']) {
+    aggregates[`aggregated-${metricName}`] = 0;
+
+    for (const input of inputs) {
+      if (!(metricName in input)) {
+        throw new AggregationError(`aggregation metric ${metricName} not found in input data`);
+      }
+      aggregates[`aggregated-${metricName}`] += parseFloat(input[metricName]);
+    };
+  }
+
+  if (AVERAGES.includes(params['aggregation-method'])) {
+    for (const metricName in aggregates) {
+      aggregates[metricName] /= inputs.length;
     }
+  }
+
+  return aggregates;
+}
+
+function _validateInputParams(inputs: ModelParams[], params: AggregationParams) {
+  if (inputs === undefined) {
+    throw new AggregationError("Input data is missing");
+  }
+
+  if (!Array.isArray(inputs)) {
+    throw new AggregationError("Input data is not an array");
+  }
+
+  if (params === undefined) {
+    throw new AggregationError("Parameters data is missing");
+  }
   
-    const aggregates: Object[] = [];
-    for (const metricName of params['aggregation-metrics']) {
-      let accumulator = 0;
-      inputs.forEach(input => {
-        if (!(metricName in input)) {
-          throw new Error(
-            'aggregation metric ' + metricName + 'not found in input data'

```

Comment:
It looks like we can use string templating (with `) here instead of string concatenation, right?